### ISSUE TITLE
HTML indentation: "g:html_indent_script1" doesn't seem to work

### ISSUE THREAD

**Issue by lkintact** [vim/vim#3205]
At `:help ft-html-indent` two variables affecting the indentation of html files are described, `g:html_indent_script1` and `g:html_indent_style1`, both taking three values: "zero", "auto" and "inc".

1. Make sure filetype indentation is on (`:filetype indent on`), put these lines into `~/vimfiles/indent/html.vim`:
```
let g:html_indent_script1 = "zero"
let g:html_indent_style1 =  "zero"
```
2. Create an html file with these contents:
```
<html>
<body>
	<style>
			div#d1 { color: red; }
div#d2 { color: green; }
     	   div#d3 { color: blue; }
	</style>
	<script>
			var v1="v1";
  var v2="v2";
   		var v3="v3";
	</script>
</body>
</html>
```
3. Open the file and hit <kbd>gg=G</kbd>, repeat for values "auto" and "inc" at step 2.
The lines inside the `<style>` block are indented consistently with the value of `g:html_indent_style1` while the lines inside the `<script>` block are always indented the same way regardless of the value of  `g:html_indent_script1` and somewhat strangely: the first line (`var v1="v1";`) is indented one more level the previous line, and the second and third ones are zero-indented.

OS: Windows 7
Gvim: 8.1.0126

**Comment by chrisbra** [comment]
Hm, the Vim script for the `<script>` tags runs into this state:
https://github.com/vim/vim/blob/44a693a1bdfe987edb649677c94c214f94b2aada/runtime/indent/html.vim#L588-L589

this comes from `indent/javascript.vim`. So CC'ing the javascript maintainer: @bounceme

**Comment by chrisbra** [comment]
To leave more context: ` GetJavascriptIndent()` always returns 0 for the javascript lines starting with `var v...;`
However the first `var v...` line is correctly indented from 
https://github.com/vim/vim/blob/44a693a1bdfe987edb649677c94c214f94b2aada/runtime/indent/html.vim#L584-L587

**Comment by bounceme** [comment]
Does the most current at pangloss/vim-javascript still fail here? I can send an update to this repo again

**Comment by chrisbra** [comment]
I checked out pangloss/vim-javascript, copied over the indent/javascript.vim over to a directory pointed to by $VIMRUNTIME. Seems to be fixed for me. Perhaps @lkintact as OP can also check.

**Comment by bounceme** [comment]
If this is indeed fixed, i give permission to take the current master from pangloss. it is stable. sorry i am kind of busy to keep checking this

**Comment by coderonline** [comment]
An attempt to help somebody on [stackoverflow](https://stackoverflow.com/a/52249799/7010506) reminded me, that I had once written a little patch, in which exactly that line appeared, which @chrisbra already mentioned and which fixed the indentation problems of multiple lines javascript within html for me:

```diff
diff --git a/runtime/indent/html.vim b/runtime/indent/html.vim
index 1a8177050..597c0d06e 100644
--- a/runtime/indent/html.vim
+++ b/runtime/indent/html.vim
@@ -587,7 +587,7 @@ func! s:Alien3()
     return eval(b:hi_js1indent)
   endif
   if b:hi_indent.scripttype == "javascript"
-    return GetJavascriptIndent()
+    return eval(b:hi_js1indent) + GetJavascriptIndent()
   else
     return -1
   endif

```

I have not investigated any further, because it worked for me and looks reasonable to me. However there may be reasons to implement something in `javascript.vim` instead.

In order to quickly test that you can use:
```vim
let g:html_indent_script1="inc" | call HtmlIndent_CheckUserSettings() | :norm ggVG=
let g:html_indent_script1="zero" | call HtmlIndent_CheckUserSettings() | :norm ggVG=
let g:html_indent_script1="auto" | call HtmlIndent_CheckUserSettings() | :norm ggVG=
```

**Comment by lkintact** [comment]
>  Perhaps @lkintact as OP can also check.

Just checked with the latest pangloss/vim-javascript, seems to be fixed for me too. Thank you guys for the help, sorry it took me that long to answer.

**Comment by lkintact** [comment]
@chrisbra not sure if I should close the issue since the latest (8.1.1622) releases at https://github.com/vim/vim-win32-installer/releases still contain a nonworking (in regard to the issue) $VIMRUNTIME/indent/javascript.vim.

**Comment by laoshaw** [comment]
this bug persists on vim 8.2

**Comment by lkintact** [comment]
Fixed for me in gVim 8.2.1551, I guess the issue can be closed.

**Comment by felipec** [comment]
This was incorrectly fixed. The fix should be in javascript, not html.

@coderonline's patch fixes some instances, but breaks others, as I mentioned in #10800 .

For example:

```html
<script>
  function test() {
      let l1;
    }
  test();
</script>
```

I've sent [a patch](https://groups.google.com/g/vim_dev/c/uL1qg42EuGs) to the mailing that back-ports the correct fix from [pangloss/vim-javascript](https://github.com/pangloss/vim-javascript) which was implemented in 2018 ([e110653](https://github.com/pangloss/vim-javascript/commit/e110653)), but for some reason they haven't been imported into vim upstream.
